pkg/karmadactl/cmdinit/bootstraptoken/agent/tlsbootstrap_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestAllowBootstrapTokensToPostCSRs(t *testing.T) {
+	tests := []struct {
+		name   string
+		client clientset.Interface
+		verify func(clientset.Interface) error
+	}{
+		{
+			name:   "AllowBootstrapTokensToPostCSRs_CreateClusterRoleBindings_Created",
+			client: fakeclientset.NewSimpleClientset(),
+			verify: func(c clientset.Interface) error {
+				clusterRoleBinding, err := c.RbacV1().ClusterRoleBindings().Get(context.TODO(), KarmadaAgentBootstrap, metav1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("unexpected error; expected cluster role binding %s to be already created, but got error: %v", KarmadaAgentBootstrap, err)
+				}
+				if clusterRoleBinding.RoleRef.Name != KarmadaAgentBootstrapperClusterRoleName {
+					return fmt.Errorf("expected clusterrolebinding ref name to be %s, but got %s", KarmadaAgentBootstrapperClusterRoleName, clusterRoleBinding.RoleRef.Name)
+				}
+				return nil
+			},
```

Comment:
Can you add a new test case for updating the ClusterRoleBinding when it already exist?

Reply:
Sounds good! Added ... waiting for GitHub CI tests to finish :)